lib/src/utils/crypto/native.dart

```diff
@@ -65,6 +78,127 @@ abstract class Cipher {
       malloc.free(mem);
     }
   }
+
+  Future<EncryptedFileInfo> encryptStream({
+    required Stream<List<int>> inputStream,
+    required File outputFile,
+    required Uint8List key, 
+    required Uint8List initialVector,
+  }) async {
+    final algorithm = getAlg(key.length * 8);
+
+    final memSize = sizeOf<IntPtr>() + key.length + initialVector.length + maxHashSize;
+    final memNeeded = malloc.call<Uint8>(memSize);
+    final intPointer = memNeeded.cast<IntPtr>();
+    final keyPointer = memNeeded.elementAt(sizeOf<IntPtr>());
+    final initialVectorPointer = keyPointer.elementAt(key.length);
+
+    final hashValuePointer = initialVectorPointer.elementAt(maxHashSize);
+    final digestAlgo = messageDigestAlgorithm.toNativeUtf8();
+
+    IOSink? outIoSink;
+    final cipherContext = EVP_CIPHER_CTX_new();
+    final mDHashContext = EVP_MD_CTX_new();
+    final digestName = EVP_get_digestbyname(digestAlgo);
+    try {
+      outIoSink = outputFile.openWrite();
+      keyPointer.asTypedList(key.length).setAll(0, key);
+      initialVectorPointer.asTypedList(initialVector.length).setAll(0, initialVector);
+      
+      EVP_EncryptInit_ex(cipherContext, algorithm, nullptr, keyPointer, initialVectorPointer);
+
+      EVP_DigestInit_ex(mDHashContext, digestName, nullptr);
+
+      await inputStream.forEach((bytes) {
```

Comment:
`foreach` should not be used with `async` functions

Reply:
<img width="747" alt="Screenshot 2023-07-28 at 13 57 58" src="https://github.com/linagora/matrix-dart-sdk/assets/43041967/07900be6-9cac-41d8-8f04-c14036a37d21">
No, forEach is Future, and inside forEach, i don't call any async function